hooks/environment

```diff
@@ -67,6 +83,12 @@ else
   secret=$(retry vault kv get -format=json "${BUILDKITE_PLUGIN_VAULT_SECRETS_PATH}" | jq -c .data)
 fi
 
+if check_buildkite_version; then
+  echo "${secret}" | buildkite-agent redactor add
```

Comment:
Is it worth making sure that this doesn't have `set -x` set above it? I.e. either running this in an additional subshell or otherwise? To avoid this secret being leaked _as part of trying to redact it_?

Reply:
How could it have a `set -x`? I am not sure what do you mean.

Reply:
I'm thinking about whether there's any way for someone to have - accidentally, or purposefully - enabled debug logging in `bash`, via `set -x`, which would then log the secret here

Reply:
You mean without a PR? Somehow injecting it into the code during runtime?

Reply:
(Maybe low risk and not something we need to worry about)

Reply:
This is an interesting conversation that I like having. However even if for example I `set +x` inthe above line, it is possible that if someone injects sth it can be injected at any time, even between my line or sth. Is there a way it can be enforced?

Reply:
I think by us adding a `set +x` (to be extra safe) then that'll cover it - I'm "thinking out loud" - I don't think I've tried this before as a way to get another plugin's output logged out further to i.e. expose secrets

Reply:
Do you think it should be added right before the line or at the start of the file?

Reply:
Either works - right before here makes the intent clearer and makes sure it's not re-added anywhere else

Reply:
Added in 8341afb07604de424ad9bd1e0880dc705022ef94